xlators/cluster/afr/src/afr-common.c

```diff
@@ -6202,6 +6202,13 @@ afr_handle_upcall_event(xlator_t *this, struct gf_upcall *upcall)
         case GF_UPCALL_CACHE_INVALIDATION:
             up_ci = (struct gf_upcall_cache_invalidation *)upcall->data;
 
+            if (AFR_IS_ARBITER_BRICK(priv, idx)) {
+                /* Skip all update involving iatt from arbiter
+                 * node, since the size contains invalid info
+                 */
+                up_ci->flags &= ~UP_ATTR_FLAGS;
+                up_ci->flags &= ~UP_PARENT_DENTRY_FLAGS;
```

Comment:
Why are we skipping UP_PARENT_DENTRY_FLAGS too?

Reply:
I'm under the impression that in normal cases we are not passing arbiter as the readable subvolume including for directory operations. If that is true, we should not be bothered to have any kind of iatt update from arbiter node. UP_PARENT_DENTRY_FLAGS  will also update the iatt in the md-cache, so any flag that requests an iatt change was removed from the upcall notification for the arbiter node.

Reply:
@itisravi `afr_readables_fill` set arbiter bricks as non-readable for both dir and non-dir. Please correct me if I'm wrong. So we should not have a case where an iatt information from arbiter is cached, hence we don't need to invalidate any iatt changes from Arbiter.

Reply:
@rafikc30 Yes, arbiter is non-readable for both data and metadata. I am thinking if we are skipping some flags, why not skip all flags. mdc_invalidate() seems to be acting on different things for other flags too like UP_XATTR, UP_XATTR_RM. I think for now we can go with this.

@pranithk @karthik-us are you planning to review the changes?

Reply:
I am going to do the review tomorrow most probably.